hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -536,7 +539,9 @@ public <R> InternalCompletableFuture<Map<K, R>> submitToKeysInternal(Set<K> keys
         } finally {
             Set<?> ncKeys = serializeKeys ? dataKeys : keys;
             for (Object key : ncKeys) {
-                invalidateNearCache(key);
+                if (!(entryProcessor instanceof ReadOnly)) {
```

Comment:
we don't need to check the entry processor in each loop iteration, we could do it once outside of the loop

Reply:
Done